internal/outputpaths/outputpaths.go

```diff
@@ -199,3 +199,19 @@ func getDeclarationEmitExtensionForPath(fileName string) string {
 	}
 	return tspath.ExtensionDts
 }
+
+// SourceFileMayBeEmitted checks if a source file may be emitted by the compiler
+func SourceFileMayBeEmitted(sourceFile *ast.SourceFile, options *core.CompilerOptions) bool {
```

Comment:
I think you misunderstood my last request; let me clarify:

1. Replace this function body with that of `sourceFileMayBeEmitted` from `internal/compiler/emitter.go`. Please copy the function body verbatim, including comments.
2. Add a `forceDtsEmit` parameter to the signature here
3. Delete `sourceFileMayBeEmitted` from `internal/compiler/emitter.go` and fix existing uses by importing and calling the one from `outputpaths`

Let’s leave the to-dos in that function alone as they are a bit out of scope for this PR.